File: core/src/main/java/cloud/vinh/rebirthsaga/game/algorithms/DungeonGenerator.java
package cloud.vinh.rebirthsaga.game.algorithms;

import cloud.vinh.rebirthsaga.game.grid.GeneratedFloor;

/** Project interface in front of dungeon generation. Implementations must be
 * deterministic for identical arguments and must return detached data owned by
 * the project (no library grids, RNGs or mutable arrays escape). Pure Java:
 * callable from a worker thread. Reachability validation and bounded retries
 * arrive with the Phase 3 movement slice. */
public interface DungeonGenerator {
    GeneratedFloor generate(int width, int height, long seed);
}

File: core/src/main/kotlin/cloud/vinh/rebirthdungeon/game/algorithms/DungeonGenerator.kt
package cloud.vinh.rebirthdungeon.game.algorithms

import cloud.vinh.rebirthdungeon.game.grid.GeneratedFloor

/** Project interface in front of dungeon generation. Implementations must be
 * deterministic for identical arguments and must return detached data owned by
 * the project (no library grids, RNGs or mutable arrays escape). Pure JVM
 * code, callable from a worker thread. FloorGeneration owns reachability
 * validation and bounded deterministic retries. */
fun interface DungeonGenerator {
    fun generate(width: Int, height: Int, seed: Long): GeneratedFloor
}
